### PR TITLE
Ecommerce detector

### PR DESCRIPTION
**Pull request by corywilson88**
Adds the following: 

1. Can detect if there is a "cart" on the current webpage you are on. This can be used as our "e-commerce detector"
2. Checks if the website you are on has SSL Certificate. This can be added to the risk weight/score equation 
3. Uses chrome/browser built-in `isSecue` API and returns a boolean called `isSecure` that can be used in the weight / score equation as well

### CODE REVIEW

File: crawl.js [hunk 1]
```diff
@@ -24,4 +24,10 @@ chrome.runtime.onMessage.addListener(function (message) {
     console.log("Website: " + title);
     console.log(htmlDocument);
   }
+
+  if (htmlDocument.documentElement.innerHTML.includes("cart")) {
+    console.log("Found the word 'cart' in the HTML document");
+  } else {
+    console.log("No cart found");
```
Comment (corywilson88) [inline crawl.js:31]: this is where we can add the label of 'ecommerce' to a site

File: src/background/background.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 //TODO add back in "use strict";
 
+const debug = true;
```
Comment (corywilson88) [inline src/background/background.js:3]: I can remove this if wanted. I was just testing something that had nothing to do with WOT so this stopped those calls from being made.

File: src/background/background.js [hunk 3]
```diff
@@ -134,10 +136,58 @@ browser.webNavigation.onCompleted.addListener(function (details) {
...
+        data = json;
+      });
+    }
+  }
+});
+
+// Checks for SSL Certificate on website
+// TODO: do not check on brand new tab
+// TODO: save to local storage?
+browser.tabs.onUpdated.addListener(async (tabId, changeInfo, tab) => {
```
Comment (corywilson88) [inline src/background/background.js:152]: Checks for SSL Certificates

File: src/background/background.js [hunk 4]
```diff
@@ -134,10 +136,58 @@ browser.webNavigation.onCompleted.addListener(function (details) {
...
+        new Uint8Array(certificate)
+      )
+        .map((b) => b.toString(16).padStart(2, "0"))
+        .join("")}`
+    );
+  }
+
+  // Only run the code if the URL has changed and is not a blank page
+  // Checks getSecurityInfo from browser/chrome API
+  if (changeInfo.url && changeInfo.url !== "about:blank") {
```
Comment (corywilson88) [inline src/background/background.js:173]: grabs `securityInfo` from browser/chrome API

**Review by thrandale — APPROVED** [review]

**Pull request merged**